solutions/CF Garland.mdx

```diff
@@ -2,10 +2,78 @@
 id: cf-garland
 source: CF
 title: An impassioned circulation of affection
-author: Neo Wang
+author: Neo Wang, Qi Wang
 ---
 
-[Official Editorial](https://codeforces.com/blog/entry/52449)
+[Official DP Editorial](https://codeforces.com/blog/entry/52449)
+
+## Solution (Two Pointer)
+
+**Time Complexity:** $\mathcal{O}(nq)$
+
+We can keep a sliding window of maximum consecutive runs of that letter with a maximum of $m$ letters exchanged.
+
+<LanguageSection>
+
+<CPPSection>
+
+### C++ Implementation:
+
+```cpp
+#include <bits/stdc++.h> // see /general/running-code-locally
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sz(x) (int)(x).size()
+using pi = pair<int,int>;
+#define f first
+#define s second
+#define mp make_pair
+void setIO(string name = "") {
+	cin.tie(0)->sync_with_stdio(0); // see /general/fast-io
+	if (sz(name)) {
+		freopen((name+".in").c_str(), "r", stdin); // see /general/io
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+int N, T;
+string W;
+
+int main() {
+	setIO();
+	cin >> N >> W >> T;
+
+	for(int i = 0; i < T; i++){
+		int a, m = 0; char b;
+		cin >> a >> b;
+		
+		int l, r;
+		l = r = 0;
+		while(l < N && r < N) {
+			while(r < N) {
+				if(W[r] != b) {
+					if(a == 0) break;
+					a--;
+				}
+				r++;
+			}
+
+			m = max(m, r - l);
+
+			a += W[l++] == b ? 0 : 1;
```

Comment:
```suggestion
			a += !W[l++] == b
```
could also do this ^ if you want :p

Reply:
! comes before == in operator precedence

Reply:
Ehhh idt this matters lol

Reply:
think it's fine as is tho

Reply:
yeah me too